git/pre-commit

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+set -eu
+
+# This file will exist in $project_root/.git/hooks/, so we need to move back up
+# to $project_root. This apparent mess needs to happen because macOS doesn't
+# have `readlink -f`.
+declare src="$0"
+declare dir=''
+
+while [ -h "$src" ]; do
+  dir="$(cd -P "$( dirname "$src")" && pwd)"
+  src="$(readlink "$src")"
+  [[ "$src" != /* ]] && src="$dir/$src"
+done
+
+cd -P "$(dirname "$src")/../../"
+
+##### This is the actual pre-commit hook. #####
+
+# If the developer has a virtual environment in the canonical location, then
+# this will attempt to use that. Otherwise, assume the necessary things are
+# installed globally.
+declare -r venv='./venv/bin/activate'
+if [ -f "$venv" ]; then
+  source "$venv"
+fi
```

Comment:
I appreciate the automagic here, but for the record I do _not_ use that path for my development virtualenv. I _do_ use that path when testing inside Tails, for use with `securedrop-admin`. But on my host machine, I use `virtualenvwrapper`, so the actual virtualenv files are stored under `~/.virtualenvs`.

I recommend wrapping all of this virtualenv logic in an if conditional that skips it if the environment variable `VIRTUAL_ENV` is defined, which it will be if a virtualenv is already active.

Reply:
I added the logic about the env-var, but left the `venv` logic since it's still useful.